### ISSUE TITLE
Is the "default" label provided for flag help text meant to show "stock" defaults or provided flag values?

### ISSUE THREAD

**Issue by atc0005** [alexflint/go-arg#92]
I'm new to Go, so my apologies if I'm off-base and have an obvious user error here.

---

I've configured a configuration option to have some default values via the "old way" (in reference to #91) and when not providing command-line flag values the `[default: VALUE]` labels are showing as expected.

However, when I use a method to validate the received flag values against a set of allowed values, return an error and then let the caller handle it by calling `WriteHelp(os.Stdout)` I see the provided flag values listed as defaults.

Example (severely truncated) output shown when calling the application with `-h` flag:

```ShellSession
  --console-output CONSOLE-OUTPUT
                         Specify how log messages are logged to the console. [default: stdout]
```

When calling `WriteHelp(os.Stdout)` after failing validation due to an unsupported flag value (again with heavily truncated output):

```ShellSession
configuration validation failed: invalid option "tacos" provided for console output destination

  --console-output CONSOLE-OUTPUT
                         Specify how log messages are logged to the console. [default: tacos]
```

In this case "tacos" isn't the *default* value, but rather what was chosen. This might be too nitpicky to worry with, but as a new user of the package it stood out to me and I wanted to mention it just in case this was unintended behavior.

**Comment by alexflint** [comment]
Hey @atc0005 thanks for reporting this. The behavior you're seeing here is not intended - it's a bug. I'll try to clear this up in #91

**Comment by alexflint** [comment]
Fixed in #91

**Comment by atc0005** [comment]
> Fixed in #91

Thanks!
